lib/sidekiq/instrument/middleware/client.rb

```diff
@@ -8,10 +8,13 @@ def call(worker_class, job, queue, redis_pool)
       # worker_class is a const in sidekiq >= 6.x
       klass = Object.const_get(worker_class.to_s)
       class_instance = klass.new
+
+      result = yield
```

Comment:
Not sure how to write a test for this. Tried using `.ordered` in assertions but it never failed when moving `result = yield` to after the `Statters`'s.

Reply:
Just `raise` in the yielded block and then assert that the `Statter` never got called. Presumably that's why you would want this functionality anyway?

Reply:
Yes, we want to move `yield` up because sometimes enqueue vs. dequeue stats are off by a large number.